File: src/ProjectR/content/scripts/spells/MiraclePoke.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Miracle Poke"
end

function GetDescription()
    return "Poke an ally, increasing all their stats."
end

function IsSupportSpell()
    return true
end

function GetMPCost()
    return .48
end

function GetDelay()
  return .25
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  for i = HP, SIL do
    d:BuffStat(i, .35)
  end
end

File: src/ProjectR.Scripting/content/scripts/spells/MiraclePoke.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class MiraclePoke : SpellScriptBase
    {
        public override string Name { get { return "Miracle Poke"; } }
        public override string Description { get { return "Poke an ally, increasing all their stats."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return true; } }

        public override double MPCost { get { return 48; } }
        public override double Delay { get { return .25; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            BuffAllStats(target, .35);
        }
    }
}